core/src/main/java/com/linecorp/armeria/server/RoutingTrieBuilder.java

```diff
@@ -137,8 +136,10 @@ private void addRoute(NodeBuilder<V> node, Entry<V> entry) {
 
             // Count the number of characters having the same prefix.
             int same = 0;
-            while (same < max && p.charAt(same) == path.charAt(same)) {
+            final PathReader pathReader = new PathReader(path);
+            while (same < max && pathReader.isSameChar(p.charAt(same))) {
                 same++;
+                pathReader.moveIndex();
```

Comment:
How about setting the index below the `while`?
```java
while (same < max && pathReader.isSameChar(p.charAt(same))) {
    same++;
}
pathReader.setIndex(same);
```

Reply:
`pathReader.isSameChar()` uses `index` so the `index` has to be updated in `while` loop.

Reply:
I missed it. 😓 We can probably pass the `same` to the `isSameChar` as the index but also fine with the current implementation. 👍